scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/FileSystemService.java

```diff
@@ -146,4 +145,37 @@ public void createFileInDirectory(String username, String type, String id, Strin
         }
     }
 
+    @Override
+    public void createZipFile(String username, String type, String id, String version)
+            throws FileSystemStorageException, IOException {
+
+        Path pathOfZip = createPath(username, type, id, version);
+        if (pathOfZip != null) {
+            String directoryOfZip = pathOfZip.toString();
+            String zipFileName = directoryOfZip + File.separator + type + "_" + id + "_version_" +
+                    version + ".zip";
+            FileOutputStream fileOutputStream = new FileOutputStream(zipFileName);
+            ZipOutputStream zipOutputStream = new ZipOutputStream(fileOutputStream);
+            File fileToZip = new File(directoryOfZip);
+
+            File[] children = fileToZip.listFiles();
+            assert children != null;
```

Comment:
I believe assertions are disabled by default, so this code wouldn't do anything once deployed. I would just check for null and throw an exception.